src/wp-includes/blocks/post-comments-form.php

```diff
@@ -72,7 +72,7 @@ function register_block_core_post_comments_form() {
  */
 function post_comments_form_block_form_defaults( $fields ) {
 	if ( wp_is_block_theme() ) {
-		$fields['submit_button'] = '<input name="%1$s" type="submit" id="%2$s" class="%3$s wp-block-button__link" value="%4$s" />';
+		$fields['submit_button'] = '<input name="%1$s" type="submit" id="%2$s" class="wp-block-button__link ' . WP_Theme_JSON_Gutenberg::get_element_class_name( 'button' ) . '" value="%4$s" />';
```

Comment:
`WP_Theme_JSON_Gutenberg::get_element_class_name` was introduced in GB's [`lib/compat/wordpress-6.1`](https://github.com/WordPress/gutenberg/blob/132b86cd48e4f785a249175476302c66dc95ed2c/lib/compat/wordpress-6.1/class-wp-theme-json-6-1.php#L118-L129).

Reply:
There is no class like that in WP core, so we will need to rename it in Gutenberg and workaround in a different way.

Reply:
Is there a reason that this should not be simply hardcoded? Looking at the source, the values are hardcoded [here](https://github.com/WordPress/gutenberg/blob/132b86cd48e4f785a249175476302c66dc95ed2c/lib/compat/wordpress-6.1/class-wp-theme-json-6-1.php#L104-L107) and then referenced in [here](https://github.com/WordPress/gutenberg/blob/132b86cd48e4f785a249175476302c66dc95ed2c/lib/compat/wordpress-6.1/class-wp-theme-json-6-1.php#L127-L129).

So, this could become simply:

```suggestion
		$fields['submit_button'] = '<input name="%1$s" type="submit" id="%2$s" class="wp-block-button__link ' . 'wp-element-button' . '" value="%4$s" />';
```

Reply:
I don't really know, but FWIW, there was a Gutenberg PR that introduced that indirection: https://github.com/WordPress/gutenberg/pull/41753.

Reply:
FWIW, I think that we might encounter other functions that are added to `WP_Theme_JSON_Gutenberg`, and for which we also might want to have a solution ready.

I've noticed that `get_element_class_name` is specifically defined in `WP_Theme_JSON_6_1`, which is located in [`class-wp-theme-json-6-1.php`](https://github.com/WordPress/gutenberg/blob/72c4666730528d27a0648958d2c8baf5d2034d5b/lib/compat/wordpress-6.1/class-wp-theme-json-6-1.php). (The `WP_Theme_JSON_6_1` class is then "aliased" as `WP_Theme_JSON_Gutenberg` in [`class-wp-theme-json-gutenberg.php`](https://github.com/WordPress/gutenberg/blob/72c4666730528d27a0648958d2c8baf5d2034d5b/lib/experimental/class-wp-theme-json-gutenberg.php).)

The existence of this versioning-and-aliasing mechanism seems to indicate for me that we can follow as similar pattern as in [#43779](https://github.com/WordPress/gutenberg/pull/43779) (and https://github.com/WordPress/wordpress-develop/pull/3158). I.e.:

- `wordpress-develop`: Backport `get_element_class_name` into Core's [`WP_Theme_JSON`](https://github.com/WordPress/wordpress-develop/blob/37b2a7fb14df0911836e63960e0ad263232aa82e/src/wp-includes/class-wp-theme-json.php).
- Gutenberg: Check for the existence of `WP_Theme_JSON::get_element_class_name` (or alternatively, for WP version >= 6.1). If it exists, use it; otherwise, fall back to `WP_Theme_JSON_Gutenberg`.

Reply:
To my knowledge, there is only one place in WordPress core where the functionality is guarded with a check for the plugin:

https://github.com/WordPress/wordpress-develop/blob/37b2a7fb14df0911836e63960e0ad263232aa82e/src/wp-includes/blocks/navigation.php#L10

I don't think it's a good practice to introduce any check for the functionality present in the Gutenberg plugin or a given WordPress version like you described. In this case, we could also simplify the code by introducing a more developer-friendly function name like `wp_theme_element_class_name` or even inlining it for WP 6.1 release if the API is still not stable enough.

Reply:
> We could also simplify the code by introducing a more developer-friendly function name like `wp_theme_element_class_name` 

Makes sense -- I also now noticed the [comment in `class-wp-theme-json.php`](https://github.com/WordPress/wordpress-develop/blob/37b2a7fb14df0911836e63960e0ad263232aa82e/src/wp-includes/class-wp-theme-json.php#L13-L15) (and its spinoffs in Gutenberg):

> This class is for internal core usage and is not supposed to be used by extenders (plugins and/or themes).
> This is a low-level API that may need to do breaking changes. Please,
> use get_global_settings, get_global_styles, and get_global_stylesheet instead.

so that's consistent with what you're suggesting 😄 

> or even inlining it for WP 6.1 release if the API is still not stable enough.

Heh, which is what [Michal was suggesting](https://github.com/WordPress/wordpress-develop/pull/3154#discussion_r962079543) 😅 

If we do that, we'd only inline it in Core (but keep the function in Gutenberg), right? 🤔

Reply:
> If we do that, we'd only inline it in Core (but keep the function in Gutenberg), right? 🤔

Yes, but we still would have to apply those changes in the Gutenberg repository because every execution of `npm run build` would override changes with the code from npm.

Reply:
Ah right, this is all dynamic blocks code 😅 But that means we wouldn't be keeping the function in Gutenberg -- or at least not the function _calls_. And since the only calls are in dynamic blocks, we'd end up with a class method that's nowhere used. Since we'd probably like to avoid that, it seems like inlining the function in Core would imply inlining it in Gutenberg, _and_ removing the function definition there, no?

Reply:
You can also change it in Gutenebrg 14.1 and revert just after that 🤷🏻‍♂️

Reply:
Update: @MaggieCabrera has thankfully started backporting the Elements API (including `get_element_class_name`): https://github.com/WordPress/wordpress-develop/pull/3206

Reply:
So I guess we can piggie-back on that PR and add an alias such as `wp_theme_element_class_name`.

Reply:
Yes, that would be nicer 👍🏻

Reply:
Created a PR for the Gutenberg part. #44099

Reply:
Thanks for wrangling the `wp_theme_element_class_name` thing. I can confirm that `WP_Theme_JSON` is not for public use (that's why we have the public wrappers for consumers to use).